src/main/java/com/powsybl/openloadflow/network/LfZeroImpedanceNetwork.java

```diff
@@ -0,0 +1,215 @@
+        for (List<VoltageControl<?>> voltageControls : voltageControlsByType.values()) {
+            if (voltageControls.size() > 1) {
+                // we take the highest target voltage (why not...) and in case of equality the voltage control
+                // with the first controlled bus ID by alpha sort
+                voltageControls.sort(Comparator.<VoltageControl<?>>comparingDouble(VoltageControl::getTargetValue)
+                        .reversed()
+                        .thenComparing(o -> o.getControlledBus().getId()));
+                VoltageControl<?> mainVc = voltageControls.get(0);
+                // first one is main, the other have are dependents
+                for (int i = 1; i < voltageControls.size(); i++) {
+                    VoltageControl<?> vc = voltageControls.get(i);
+                    vc.mergeStatus = VoltageControl.MergeStatus.DEPENDENT;
+                    linkVoltageControls(mainVc, vc);
+                }
+            }
+        }
+    }
+
+    public void tryToSplit() {
+        List<LfBranch> disabledBranches = graph.edgeSet().stream()
+                .filter(LfElement::isDisabled)
+                .collect(Collectors.toList());
+        if (!disabledBranches.isEmpty()) {
+            for (LfBranch branch : disabledBranches) {
+                graph.removeEdge(branch);
+            }
+            List<Set<LfBus>> connectedSets = new ConnectivityInspector<>(graph).connectedSets();
+            if (connectedSets.size() > 1) { // real split
+                for (LfBranch branch : disabledBranches) {
+                    branch.setSpanningTreeEdge(dc, false);
+                }
+                Set<LfZeroImpedanceNetwork> zeroImpedanceNetworks = network.getZeroImpedanceNetworks(dc);
+                zeroImpedanceNetworks.remove(this);
+                List<LfZeroImpedanceNetwork> splitZns = new ArrayList<>(2);
+                for (Set<LfBus> connectedSet : connectedSets) {
+                    var subGraph = new AsSubgraph<>(graph, connectedSet);
+                    splitZns.add(new LfZeroImpedanceNetwork(network, dc, subGraph));
+                }
+                zeroImpedanceNetworks.addAll(splitZns);
+
+                // update voltage control merge status
+                if (!dc) {
+                    for (LfZeroImpedanceNetwork splitZn : splitZns) {
+                        splitZn.updateVoltageControlMergeStatus();
+                    }
+                }
+
+                for (LfNetworkListener listener : network.getListeners()) {
+                    listener.onZeroImpedanceNetworkSplit(this, splitZns, dc);
+                }
+            } else {
+                boolean atLeastOneOfDisablingBranchIsPartOfSpanningTree = disabledBranches.stream()
+                        .anyMatch(branch -> branch.isSpanningTreeEdge(dc));
+                if (atLeastOneOfDisablingBranchIsPartOfSpanningTree) {
+                    // just update the spanning
+                    updateSpanningTree();
```

Comment:
Here you should also remove the disabledBranches from the spanning tree. They've been removed from the graph and therefore are not updated in `updateSpanningTree()`.
```suggestion
                    updateSpanningTree();
                    for (LfBranch branch : disabledBranches) {
                        branch.setSpanningTreeEdge(dc, false);
                    }
```